scipy/signal/_peak_finding.py

```diff
@@ -246,6 +250,107 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
     return np.nonzero(results)
 
 
+def _handle_arg_x(value):
+    """Check and normalize argument `x`.
+
+    See descriptions in functions `find_peaks`, `peak_prominences` and
+    `peak_widths`.
+
+    Parameters
+    ----------
+    value : array_like
+        A one-dimensional array.
+
+    Returns
+    -------
+    value : ndarray
+        A one-dimensional C-contiguous array with dtype('float64').
+    """
+    value = np.asarray(value, order='C', dtype=np.float64)
```

Comment:
Simply because this argument is passed to Cython functions that expect it to be C-Contiguous. This allowed these functions to be faster for most use cases. With the old context this was clearer.

Reply:
Ouch. I completely overlooked the Cython detail. Sorry about that.